cmd/cluster/validatepullsecret.go

```diff
@@ -11,93 +17,233 @@ import (
+
+In case the managed-script fails, --elevate can be added to get the pull-secret in the cluster directly without a managed-script.
+`,
 		Args:              cobra.ExactArgs(1),
 		DisableAutoGenTag: true,
 		Run: func(cmd *cobra.Command, args []string) {
-			cmdutil.CheckErr(ValidatePullSecret(args[0], kubeCli, flags))
+			cmdutil.CheckErr(ops.run(args[0], kubeCli, flags))
 		},
 	}
+	validatePullSecretCmd.Flags().BoolVar(&ops.elevate, "elevate", false, "get pull-secret with backplane-cluster-admin without running a managed-script")
+
+	return validatePullSecretCmd
 }
 
-func ValidatePullSecret(clusterID string, kubeCli client.Client, flags *genericclioptions.ConfigFlags) error {
-	ocm, err := utils.CreateConnection()
-	if err != nil {
-		return err
-	}
-	defer func() {
-		if ocmCloseErr := ocm.Close(); ocmCloseErr != nil {
-			_, _ = fmt.Fprintf(os.Stderr, "Cannot close the ocm (possible memory leak): %q", ocmCloseErr)
-		}
-	}()
+func newValidatePullSecretOptions() *validatePullSecretOptions {
+	return &validatePullSecretOptions{}
+}
 
-	subscription, err := utils.GetSubscription(ocm, clusterID)
-	if err != nil {
-		return err
-	}
+func (o *validatePullSecretOptions) run(clusterID string, kubeCli client.Client, flags *genericclioptions.ConfigFlags) error {
 
-	account, err := utils.GetAccount(ocm, subscription.Creator().ID())
+	// get the pull secret in OCM
+	emailOCM, err, done := getPullSecretFromOCM(clusterID)
 	if err != nil {
 		return err
 	}
+	if done {
+		return nil
+	}
 
-	registryCredentials, err := utils.GetRegistryCredentials(ocm, account.ID())
+	// get the pull secret in cluster
+	emailCluster := ""
+	if o.elevate {
+		emailCluster, err, done = getPullSecretFromClusterElevated(clusterID, kubeCli, flags)
+	} else {
+		emailCluster, err, done = getPullSecretFromCluster(clusterID)
+	}
 	if err != nil {
 		return err
 	}
-	if len(registryCredentials) == 0 {
-		_, _ = fmt.Fprintln(os.Stderr, "There is no pull secret in OCM. Sending service log.")
+	if done {
+		return nil
+	}
+
+	if emailOCM != emailCluster {
+		_, _ = fmt.Fprintln(os.Stderr, "Pull secret email doesn't match OCM user email. Sending service log.")
 		postCmd := servicelog.PostCmdOptions{
-			Template:       "https://raw.githubusercontent.com/openshift/managed-notifications/master/osd/update_pull_secret.json",
-			TemplateParams: []string{"REGISTRY=registry.redhat.io"},
-			ClusterId:      clusterID,
+			Template:  "https://raw.githubusercontent.com/openshift/managed-notifications/master/osd/pull_secret_user_mismatch.json",
+			ClusterId: clusterID,
 		}
 		if err = postCmd.Run(); err != nil {
 			return err
 		}
 		return nil
 	}
 
+	fmt.Println("Email addresses match.")
+	return nil
+}
+
+// getPullSecretFromCluster runs a managed-script to get the pull-secret
+// email from cluster without backplane elevation
+// it returns the email, error and done
+// done means a service log has been sent
```

Comment:
Consider naming the return values to make this clearer

Reply:
In the new commit, I named it `sentSL` to specifically indicate a SL had been sent.